File: src/main/java/com/saomc/util/J8String.java
package com.saomc.util;

import java.util.stream.Stream;

public final class J8String {

    private J8String() {
    }

    public static String join(String s0, String... s1) {
        final StringBuilder builder = new StringBuilder(s0);
        Stream.of(s1).forEach(s -> builder.append(s).append(s0));

        return builder.toString().trim();
    }

}

File: src/main/java/com/saomc/saoui/util/J8String.kt
package com.saomc.saoui.util

import java.util.stream.Stream

object J8String {

    fun join(s0: String, vararg s1: String): String {
        val builder = StringBuilder(s0)
        Stream.of(*s1).forEach { s -> builder.append(s).append(s0) }

        return builder.toString().trim { it <= ' ' }
    }

}
